### ISSUE TITLE
Unable to build project help please.

### ISSUE THREAD

**Issue by MoeMonsuta** [mozq/dencode-web#5]
First I love your dencode, it's what I really needed badly to convert UTF-16LE to Hex to work on some tables in Cheat Engine. and works perfectly and quickly. I was trying to build it but I keep getting this error. I'm not an advanced user and never used "Grandle" before, but I have built applications in Github with no issues. I'm really curious how I can get dencode working offline because I want to learn the trick to building it without getting this error:

```
dencode-web>gradlew appRunStage
> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> error: invalid source release: 17

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.4/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1s
1 actionable task: 1 executed
```

I have Java SDK 17 and 21 installed on my computer so I have no idea what I'm doing wrong. I set my Java bin directory path in Environmental Variables too. Any help so I can get your awesome dencode working for me locally?

**Comment by mozq** [comment]
Thank you for loving DenCode.
I've pushed new build scripts for Gradle. (Commit; d4220e5f1a1f6fd44cff59e367bf4a3442b8e076)
Could you clone the latest one and try the command `./gradlew appRunStage` again?

**Comment by mozq** [comment]
If it does not resolve, please check the Java version. `java --version`

I also found the following information about OVERRIDE_JAVA_VERSION environment variable, but I'm not sure if it will work.
https://ionic.zendesk.com/hc/en-us/articles/14973678137111-error-invalid-source-release-17

**Comment by MoeMonsuta** [comment]
> If it does not resolve, please check the Java version. `java --version`
> 
> I also found the following information about OVERRIDE_JAVA_VERSION environment variable, but I'm not sure if it will work. https://ionic.zendesk.com/hc/en-us/articles/14973678137111-error-invalid-source-release-17

Thank you for your reply. I still wasn't able to build it unfortunately with the new clone. This was the result:

```
dencode-web> ./gradlew appRunStage
Downloading https://services.gradle.org/distributions/gradle-8.4-bin.zip
............10%............20%.............30%............40%.............50%............60%.............70%............80%.............90%............100%

Welcome to Gradle 8.4!

Here are the highlights of this release:
 - Compiling and testing with Java 21
 - Faster Java compilation on Windows
 - Role focused dependency configurations creation

For more details see https://docs.gradle.org/8.4/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)
> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> error: invalid source release: 17

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 29s
1 actionable task: 1 executed
```

My Java seems to be reporting as 21. 

```
dencode-web> java --version
java 21.0.1 2023-10-17 LTS
Java(TM) SE Runtime Environment (build 21.0.1+12-LTS-29)
Java HotSpot(TM) 64-Bit Server VM (build 21.0.1+12-LTS-29, mixed mode, sharing)
```

**Comment by mozq** [comment]
On macOS, it worked with both Java17 and Java21 versions.
Is it possible to run it with Java17?

set JAVA_HOME
java —version
set JAVA_HOME= **PLEASE SET YOUR JAVA17 HOME**
java —version
./gradlew appRunStage

**Comment by mozq** [comment]
I can build with Oracle Java21 on Windows10. I used the following JDK.

Windows - x64 Compressed Archive
https://download.oracle.com/java/21/latest/jdk-21_windows-x64_bin.zip

**Comment by mozq** [comment]
> unmappable character (0x81) for encoding windows-31j
Are you using Japanese Windows? It seems that the JDK reads the source codes as Windows-31j (Shift_JIS) encoding.
Let me check the encoding configuration.

**Comment by MoeMonsuta** [comment]
Yes I am using Japanese system locale.

I tried to compile with Java sdk 17 and 21 they both failed.

**Comment by mozq** [comment]
I've pushed latest build.gradle file (Commit; 19e9514177a6501cd6505f1583683ce66c9c20d6). Please try.

**Comment by MoeMonsuta** [comment]
Thanks again, it seemed to not get errors this time. It is locked at this part now:

```
gradlew appRunStage

> Task :appRunStage
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
<===========--> 91% EXECUTING [1m 35s]
> :appRunStage

```
Is there something else I must do?

**Comment by mozq** [comment]
どういたしまして。
Please ignore the notice. You just access to http://localhost:8080/ . That's it ;)

If you want to stop the local server, please use [CTRL+C] key.

**Comment by MoeMonsuta** [comment]
Oh yes, thank you. I didn't know I had to run the localhost on port 8080! It seems to be working now, but the alignment seems off. When I try to click on the decoded address for example it hides the entire line, but on your main website it opens a nice wide box.

Mine:
![2023-10-23_031427](https://github.com/mozq/dencode-web/assets/27302728/5347e970-1b41-4d03-979f-21871369cd83)

Yours:
![2023-10-23_031608](https://github.com/mozq/dencode-web/assets/27302728/697602bc-b0f5-43a4-8038-3bdbea792a9c)

I have tried this in Firefox and Chrome.
